File: backend/app/schemas/cost_budget.py
from typing import Optional
import uuid
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel, Field


class CostBudgetCreate(BaseModel):
    monthly_limit: Decimal = Field(..., ge=0)


class CostBudgetUpdate(BaseModel):
    monthly_limit: Optional[Decimal] = Field(default=None, ge=0)


class CostBudgetResponse(BaseModel):
    id: uuid.UUID
    monthly_limit: float
    current_usage: float
    month: str
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class BudgetCheckResponse(BaseModel):
    allowed: bool
    remaining: float
    limit: float
    usage: float


class UsageHistoryItem(BaseModel):
    month: str
    usage: float
    limit: float

File: frontend/src/api/costBudget.ts
import client from './client';

export interface CostBudget {
  id: string;
  monthly_limit: number;
  current_usage: number;
  month: string;
  created_at: string;
  updated_at: string;
}

export interface BudgetCheck {
  allowed: boolean;
  remaining: number;
  limit: number;
  usage: number;
}

export interface UsageHistoryItem {
  month: string;
  usage: number;
  limit: number;
}

export interface CostBudgetUpdate {
  monthly_limit: number;
}

export const costBudgetApi = {
  getCurrent: () => client.get<CostBudget>('/cost-budget'),

  update: (data: CostBudgetUpdate) => client.put<CostBudget>('/cost-budget', data),

  check: () => client.get<BudgetCheck>('/cost-budget/check'),

  getHistory: (months?: number) =>
    client.get<UsageHistoryItem[]>('/cost-budget/history', { params: months ? { months } : {} }),
};
